sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -48,6 +48,8 @@ case class If(predicate: Expression, trueValue: Expression, falseValue: Expressi
   override def second: Expression = trueValue
   override def third: Expression = falseValue
   override def nullable: Boolean = trueValue.nullable || falseValue.nullable
+  override def foldable: Boolean = children.forall(_.foldable)
```

Comment:
I think this is applied for all `ConditionalExpression`s?

Reply:
+1, pull out to `ConditionalExpressions`